server: TestClusterVersionUnreplicatedRaftTruncatedState failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/8892e379d84a36b29003420189edd1e10db41d71

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestClusterVersionUnreplicatedRaftTruncatedState PKG=github.com/cockroachdb/cockroach/pkg/server TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1329191&tab=buildLog

```
I190607 08:21:27.539378 56353 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] transport racer
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190607 08:21:27.539751 56353 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] transport racer
1      [async] closedts-rangefeed-subscriber
I190607 08:21:27.540007 56353 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] transport racer
I190607 08:21:27.548229 56531 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
I190607 08:21:27.594157 56433 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:41225: context canceled
I190607 08:21:27.594283 56546 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43013: context canceled
I190607 08:21:27.595417 56432 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33323: context canceled
I190607 08:21:27.596645 56503 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33323: context canceled
W190607 08:21:27.597242 54223 server/node.go:851  [n1,summaries] error recording status summaries: node unavailable; try another peer
I190607 08:21:27.598040 56505 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43013: context canceled
I190607 08:21:27.599491 56504 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:44213: context canceled
W190607 08:21:27.649560 54415 storage/store.go:1525  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I190607 08:21:27.656201 56565 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:41225: context canceled
I190607 08:21:27.657744 56566 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:44213: context canceled
I190607 08:21:27.658467 56567 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n4: failed to connect to n4 at 127.0.0.1:43013: context canceled
I190607 08:21:27.662084 56552 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:41225: context canceled
I190607 08:21:27.662213 56554 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n3: failed to connect to n3 at 127.0.0.1:44213: context canceled
I190607 08:21:27.663185 56553 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n2: failed to connect to n2 at 127.0.0.1:33323: context canceled
W190607 08:21:27.704036 54222 ts/db.go:214  [n1,ts-poll] node unavailable; try another peer
W190607 08:21:27.706588 54415 storage/store.go:1525  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
I190607 08:21:27.717692 54417 kv/transport_race.go:113  transport race promotion: ran 112 iterations on up to 1898 requests
I190607 08:21:27.730353 54197 storage/engine/rocksdb.go:753  closing rocksdb instance at "/tmp/TestClusterVersionUnreplicatedRaftTruncatedState439847609/0"
I190607 08:21:27.744242 54197 storage/engine/rocksdb.go:753  closing rocksdb instance at "/tmp/TestClusterVersionUnreplicatedRaftTruncatedState439847609/1"
I190607 08:21:27.763647 54197 storage/engine/rocksdb.go:753  closing rocksdb instance at "/tmp/TestClusterVersionUnreplicatedRaftTruncatedState439847609/2"
I190607 08:21:27.778116 54197 storage/engine/rocksdb.go:753  closing rocksdb instance at "/tmp/TestClusterVersionUnreplicatedRaftTruncatedState439847609/3"
--- FAIL: TestClusterVersionUnreplicatedRaftTruncatedState (20.73s)
    version_cluster_test.go:341: getting term: requested entry at index is unavailable

```

--- a/pkg/server/version_cluster_test.go
+++ b/pkg/server/version_cluster_test.go
@@ -336,8 +336,12 @@ ALTER TABLE kv SPLIT AT SELECT i FROM generate_series(1, 9) AS g(i);
 							var ba roachpb.BatchRequest
 							ba.RangeID = r.RangeID
 							ba.Add(truncate)
-							if _, err := s.DB().NonTransactionalSender().Send(ctx, ba); err != nil {
-								t.Fatal(err)
+							if _, pErr := s.DB().NonTransactionalSender().Send(ctx, ba); pErr != nil {
+								// The index may no longer be around if the system decided to truncate the
+								// logs after we read it. Don't fail the test when that happens.
+								if !testutils.IsPError(pErr, "requested entry at index is unavailable") {
+									t.Fatal(err)
+								}
 							}
 						}
 						return true // want more

